src/main/java/marites/Marites.java

```diff
@@ -0,0 +1,312 @@
+           "Alright, task added:%n  %s%nYour list currently has %d tasks.\n";
+    private static final String ADD_TASK_MISSING_PARAMETER_MESSAGE =
+           "You're missing a parameter: %s\n";
+    private static final String ADD_TASK_MISSING_DESCRIPTION_MESSAGE =
+            "Please add a description to your task.\n";
+    private static final String ADD_TASK_UNKNOWN_TASK_MESSAGE =
+            "I don't know this task type: '%s'\n";
+    private static final String MARK_DONE_MESSAGE = "Good job on getting this done!";
+    private static final String MARK_UNDONE_MESSAGE = "Okay, I've marked this as not yet done:";
+    private static final String DELETE_TASK_MESSAGE =
+            "Alright, task deleted:%n  %s%nYour list currently has %d tasks.\n";
+    private static final String INVALID_TASK_NUMBER_MESSAGE =
+            "I either didn't get a task number, or couldn't understand it: '%s'\n";
+    private static final String SET_TASK_STATUS_FORMAT_STRING = "%s%n    %s\n";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+    public static final String SAVE_FILENAME = "tasklist.ser";
+    public static final String READ_TASK_LIST_ERROR_MESSAGE = "WARNING: Error when reading saved task list; initializing with empty list";
+    public static final String SAVE_TASK_LIST_ERROR_MESSAGE = "WARNING: Error occurred while saving task list";
+
+    /** The task list being tracked by marites.Marites. */
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+
+
+    public static void main(String[] args) {
+        initializeTaskList();
+        printIntroduction();
+        while (true) {
+            String userInput = SCANNER.nextLine();
+            String commandFeedback = processUserCommand(userInput);
+            showFeedback(commandFeedback);
+        }
+    }
+
+    private static void initializeTaskList() {
+        File saveFile = new File(SAVE_FILENAME);
+        if (!saveFile.exists()) {
+            tasks = new ArrayList<>();
+            return;
+        }
+        try {
+            FileInputStream fileIn = new FileInputStream(SAVE_FILENAME);
+            ObjectInputStream objIn = new ObjectInputStream(fileIn);
+            tasks = (ArrayList<Task>)objIn.readObject();
+        } catch (IOException | ClassNotFoundException | ClassCastException e) {
+            System.out.println(READ_TASK_LIST_ERROR_MESSAGE);
+            tasks = new ArrayList<>();
+        }
+    }
+
+    private static void saveTaskList() {
+        try {
+            File saveFile = new File(SAVE_FILENAME);
+            saveFile.createNewFile();
+            FileOutputStream fileOut = new FileOutputStream(SAVE_FILENAME);
+            ObjectOutputStream objOut = new ObjectOutputStream(fileOut);
+            objOut.writeObject(tasks);
+        } catch (IOException e) {
+            System.out.println(SAVE_TASK_LIST_ERROR_MESSAGE);
+        }
+    }
+
+    /* OUTPUT */
+
+    /**
+     * Prints a separator string.
+     */
```

Comment:
Your habit of documenting the code is good, but [Do Not Repeat the Obvious](https://nus-cs2113-ay2122s2.github.io/website/se-book-adapted/chapters/codeQuality.html#do-not-repeat-the-obvious)